tests/unit/test_meta_model_repurposer.py

```diff
@@ -253,7 +253,7 @@ def _test_predict(self, mock_model_handler, mock_validate_method, test_predict_p
         # Train or Load target model from file
         if self.repurposer_class == GpRepurposer:
             mock_model_handler.return_value.get_layer_output.return_value =\
-                {'l1': self.train_features}, self.train_labels
+                {'l1': self.train_features[:self.num_data_points]}, self.train_labels[:self.num_data_points]
```

Comment:
num_data_points comes from derived class. Avoid it in training and use only during predict?